pkg/registry/generic/etcd/etcd.go

```diff
@@ -285,6 +286,12 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 			if err != nil {
 				return nil, nil, err
 			}
+			if newVersion == 0 {
+				// This is hacky, we shouldn't use QualifiedResource as GroupVersionKind. NewConflict has the same problem.
```

Comment:
Apart from what the comment says, probably it's not a good idea to import the `field` package. @nikhiljindal, do you have better idea?